nms/src/main/java/com/owen1212055/biomevisuals/nms/hooks/RegistryHook.java

```diff
@@ -34,32 +36,37 @@ public <T> DataResult<T> encode(RegistryAccess input, DynamicOps<T> ops, T prefi
                 }
 
                 JsonObject mainRegistry = result.get().orThrow().getAsJsonObject();
-                // Iterate through active override types
-                for (var entry : hooks.entrySet()) {
+                // Iterate through hooked registry types
+                for (var hookedTypeKey : HookType.getKeys()) {
                     // Retrieve the registry array
-                    JsonArray registry = mainRegistry.get(entry.getKey().getKey().toString()).getAsJsonObject().getAsJsonArray("value");
+                    JsonArray registry = mainRegistry.get(hookedTypeKey.toString()).getAsJsonObject().getAsJsonArray("value");
 
-                    // Populate it and key it by its identifier, this is because they are stored in an array and this allows us to fetch them nicely.
-                    Map<String, JsonObject> registeredObjects = new HashMap<>(registry.size());
+                    // Populate maps keyed by identifiers, because they are stored in an array and this allows us to fetch them nicely.
+                    Map<NamespacedKey, JsonObject> registryEntries = new HashMap<>(registry.size());
+                    Map<NamespacedKey, JsonObject> registryEntryHolders = new HashMap<>(registry.size());
                     for (JsonElement jsonElement : registry) {
-                        registeredObjects.put(jsonElement.getAsJsonObject().get("name").getAsString(), jsonElement.getAsJsonObject().getAsJsonObject("element"));
+                        var namespacedKey = Objects.requireNonNull(NamespacedKey.fromString(jsonElement.getAsJsonObject().get("name").getAsString()));
```

Comment:
var + don't use Objects.requireNonNull.
In general your IDE probably screams at you to add this, but, it's not needed.